### ISSUE TITLE
Snapshots working, streaming not

### ISSUE THREAD

**Issue by colinuss** [hjdhjd/homebridge-unifi-protect#122]
**Describe The Problem:**
I can see the camera snapshots in the Home app on all devices, however when I click on one the streaming does not start (loading forever) and I get the following error message in the log - always and reproducible

**To Reproduce:**
See above

**Logs:**
3.10.2020, 13:19:43 Bridge is running on port 51826.
3.10.2020, 13:22:36 (node:32015) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'address' of undefined
3.10.2020, 13:22:36 at RTPStreamManagement.generateSetupEndpointResponse (/usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/camera/RTPStreamManagement.js:675:98)
3.10.2020, 13:22:36 at /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/camera/RTPStreamManagement.js:644:27
3.10.2020, 13:22:36 at /usr/local/lib/node_modules/@hoobs/hoobs/node_modules/hap-nodejs/dist/lib/util/once.js:15:25
3.10.2020, 13:22:36 at ProtectStreamingDelegate.prepareStream (/home/hoobs/.hoobs/node_modules/homebridge-unifi-protect/dist/protect-stream.js:140:9)
3.10.2020, 13:22:36 at runMicrotasks (<anonymous>)
3.10.2020, 13:22:36 at processTicksAndRejections (internal/process/task_queues.js:97:5)
3.10.2020, 13:22:36 (node:32015) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 1)
3.10.2020, 13:22:36 (node:32015) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
3.10.2020, 13:22:46 [UniFi Protect] Zuhause [UCK-G2-PLUS] Einfahrt [UVC G3 Flex]: Stopped video streaming session.

**Homebridge Configuration:**

{
    "server": {
        "port": 80,
        "origin": "*",
        "autostart": 0,
        "home_setup_id": "X-HM://0023ISYWWNN6F",
        "polling_seconds": 5
    },
    "client": {
        "default_route": "status",
        "inactive_logoff": 30,
        "theme": "hoobs-light",
        "locale": "",
        "temp_units": "celsius",
        "country_code": "DE",
        "postal_code": "40489"
    },
    "bridge": {
        "name": "Wilke",
        "port": 51826,
        "pin": "031-45-152",
        "username": "61:69:63:68:F2:1B"
    },
    "description": "",
    "ports": {},
    "accessories": [],
    "platforms": [
        {
            "platform": "free@home",
            "plugin_map": {
                "plugin_name": "homebridge-freeathome"
            },
            "name": "free@home",
            "debug": false,
            "sysIP": "192.168.2.103",
            "username": "XXX",
            "password": "XXXX"
        },
        {
            "platform": "UniFi Protect",
            "plugin_map": {
                "plugin_name": "homebridge-unifi-protect"
            },
            "controllers": [
                {
                    "address": "192.168.2.107",
                    "username": "homebridge",
                    "password": "XXXX"
                }
            ],
            "verboseFfmpeg": false
        }
    ]
}

**Screenshots:**
None

**Environment:**

* **Homebridge Version:** 
HOOBS 3 Version | 3.2.6

* **Node Version:** Node Version | 12.16.3
* **Homebridge-UniFi-Protect Plugin Version:** 3.7.8
* **Apple Device and iOS / macOS / iPadOS / tvOS Version:** all latest iOS 14 versions
* **Operating System and OS Version:** macOS Catalina

**Comment by hjdhjd** [comment]
Hi there! This issue has been asked and answered:

https://github.com/hjdhjd/homebridge-unifi-protect/issues/100#issuecomment-697056014
https://github.com/hjdhjd/homebridge-unifi-protect/issues/109#issuecomment-700233266

TL;DR: HOOBS needs to be updated. The HOOBS developers are aware. They have no timeline to update their version of Homebridge. If you want to use HOOBS with this plugin, you’ll need to downgrade to a version below v3.6.0. If streaming isn’t important to you, you can use the latest version with HOOBS, however streaming will not work - and will not work - until HOOBS updates their version of Homebridge.

**Comment by colinuss** [comment]
Thanks for your swift reply. Apologies, I am new to github and overlooked the “closed” tab for issues. Thanks for a great plugin. 

> Am 03.10.2020 um 16:33 schrieb hjdhjd <notifications@github.com>:
> 
> have

**Comment by hjdhjd** [comment]
No worries! My suggestion, either further encourage the developers for a more swift update, or a timeline for one, for HOOBS...or if you prefer, you can switch to the official Homebridge image. I suspect there's going to be more differences in the future, as Homebridge is evolving at a good pace right now, and unless HOOBS increases it's update schedule substantially, more issues are likely in the future.

To be clear - I think HOOBS is a terrific platform for those who prefer it, however, it really needs to keep up with where Homebridge is going if they want to maintain compatibility in the medium/long-term.

Best of luck!
